project-minimization-core/src/main/kotlin/org/plan/research/minimization/core/algorithm/graph/TransposedGraph.kt

```diff
@@ -0,0 +1,31 @@
+package org.plan.research.minimization.core.algorithm.graph
+
+import org.plan.research.minimization.core.model.DDItem
+import org.plan.research.minimization.core.model.graph.GraphEdge
+import org.plan.research.minimization.core.model.graph.GraphWithAdjacencyList
+
+import arrow.core.getOrNone
+
+internal class TransposedGraph<V : DDItem, E : GraphEdge<V>, G : GraphWithAdjacencyList<V, E>>(
+    val originalGraph: G,
+    override val vertices: List<V> = originalGraph.vertices.toList(),
+) : GraphWithAdjacencyList<V, TransposedGraph.TransposedEdge<V>>() {
```

Comment:
It seems to me that we can use only one generic parameter here, because others don't give additional information here

```suggestion
internal class TransposedGraph<V : DDItem>(
    val originalGraph: GraphWithAdjacencyList<V, *>,
    override val vertices: List<V> = originalGraph.vertices.toList(),
) : GraphWithAdjacencyList<V, TransposedGraph.TransposedEdge<V>>() {
```

Reply:
Nope, we still need `E` if we want to condensate all the edges to a single CondensedEdge